src/tracer.c

```diff
@@ -117,7 +117,66 @@ find(pid_t pid)
     return pinfo + i;
 }
 
-uint8_t        *hash_file(char *);
+uint8_t *hash_file(char *);
+
+static void
+handle_close(int fd, FILE_INFO *f)
+{
+    f->hash = hash_file(f->path);
+}
+
+static void
+handle_open(pid_t pid, int syscall, int fd, const unsigned long long *args)
```

Comment:
probably worth splitting `openat` to a different function.

Reply:
@zvr Thought about doing that, won't need `syscall` as an argument and perhaps it will also be more readable. In that notation we should also create one for `creat` but that would be redundant.

By the way i noticed that there is also an `openat2(2)` syscall slightly different from `openat(2)`? Should be handled as well, i think it's kind of the same that `creat` is to `open`.